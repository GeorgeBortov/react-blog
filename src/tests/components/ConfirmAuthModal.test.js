import React from 'react';
import { shallow } from 'enzyme';
import ConfirmAuthModal from '../../components/ConfirmAuthModal';

let onRequestClose, addNewAccount;

beforeEach(() => {
    onRequestClose = jest.fn();
    addNewAccount = jest.fn();
});

test('should render close ConfirmAuthModal correctly', () => {
    const wrapper = shallow(<ConfirmAuthModal
        isOpen={false}
        addNewAccount={addNewAccount}
        onRequestClose={onRequestClose}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should render open ConfirmAuthModal correctly', () => {
    const wrapper = shallow(<ConfirmAuthModal
        isOpen={true}
        addNewAccount={addNewAccount}
        onRequestClose={onRequestClose}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle addNewAccount', () => {
    const wrapper = shallow(<ConfirmAuthModal
        isOpen={false}
        addNewAccount={addNewAccount}
        onRequestClose={onRequestClose}
    />);
    wrapper.find('.removeExpense').simulate('click');
    expect(addNewAccount).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
});

test('should handle onRequestClose with button "NO" ', () => {
    const wrapper = shallow(<ConfirmAuthModal
        providerName={'google.com'}
        isOpen={false}
        addNewAccount={addNewAccount}
        onRequestClose={onRequestClose}
    />);
    wrapper.find('.button--secondary').simulate('click');
    expect(onRequestClose).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
});

test('should handle onRequestClose with button "X" ', () => {
    const wrapper = shallow(<ConfirmAuthModal
        providerName={'google.com'}
        isOpen={false}
        addNewAccount={addNewAccount}
        onRequestClose={onRequestClose}
    />);
    wrapper.find('.modal__close').simulate('click');
    expect(onRequestClose).toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
});