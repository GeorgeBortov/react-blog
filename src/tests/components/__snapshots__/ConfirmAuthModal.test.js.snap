// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle addNewAccount 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={false}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={[MockFunction]}
    />
    <h3
      className="modal__title"
    >
      Do you want to sign in with a different account?
    </h3>
    <button
      className="button removeExpense"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</Modal>
`;

exports[`should handle onRequestClose with button "NO"  1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={false}
  onRequestClose={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    />
    <h3
      className="modal__title"
    >
      Do you want to sign in with a different account?
    </h3>
    <button
      className="button removeExpense"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      No
    </button>
  </div>
</Modal>
`;

exports[`should handle onRequestClose with button "X"  1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={false}
  onRequestClose={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    />
    <h3
      className="modal__title"
    >
      Do you want to sign in with a different account?
    </h3>
    <button
      className="button removeExpense"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        }
      }
    >
      No
    </button>
  </div>
</Modal>
`;

exports[`should render close ConfirmAuthModal correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={false}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={[MockFunction]}
    />
    <h3
      className="modal__title"
    >
      Do you want to sign in with a different account?
    </h3>
    <button
      className="button removeExpense"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</Modal>
`;

exports[`should render open ConfirmAuthModal correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Confirmation Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__body"
  >
    <button
      className="modal__close"
      onClick={[MockFunction]}
    />
    <h3
      className="modal__title"
    >
      Do you want to sign in with a different account?
    </h3>
    <button
      className="button removeExpense"
      onClick={[MockFunction]}
    >
      Yes
    </button>
    <button
      className="button button--secondary"
      onClick={[MockFunction]}
    >
      No
    </button>
  </div>
</Modal>
`;
